src/relayserver/runServer.ts

```diff
@@ -1,106 +1,63 @@
 // TODO: convert to 'commander' format
 import fs from 'fs'
-import parseArgs from 'minimist'
 import Web3 from 'web3'
 import { HttpServer } from './HttpServer'
 import { RelayServer, RelayServerParams } from './RelayServer'
 import { KeyManager } from './KeyManager'
 import { TxStoreManager, TXSTORE_FILENAME } from './TxStoreManager'
-import { getRelayHubAddress } from '../cli/utils'
 import ContractInteractor from '../relayclient/ContractInteractor'
 import { configureGSN } from '../relayclient/GSNConfigurator'
+import { parseServerConfig, resolveServerConfig, ServerConfigParams } from './ServerConfigParams'
 
-export interface ServerConfigParams {
-  baseRelayFee?: number | string
-  pctRelayFee?: number | string
-  url?: string
-  port?: number | string
-  relayHubAddress?: string
-  gasPricePercent?: number | string
-  ethereumNodeUrl?: string
-  workdir?: string
-  devMode?: boolean
-  debug?: boolean
-  registrationBlockRate?: number | string
-}
-
-function error (err: string): void {
+function error (err: string): never {
   console.error(err)
   process.exit(1)
 }
 
-// use all camel-case entries from environment as defaults.
-const envDefaults = Object.entries(process.env)
-  .filter(([k]) => /^[a-z][A-Za-z]*$/.test(k))
-  .reduce((obj, [k, v]) => ({ ...obj, [k]: v }), {})
-
-const argv = parseArgs(process.argv.slice(2), {
-  string:
-    [
-      'config',
-      'baseRelayFee',
-      'pctRelayFee',
-      'url',
-      'port',
-      'relayHubAddress',
-      'gasPricePercent',
-      'ethereumNodeUrl',
-      'workdir'
-    ],
-  boolean: ['devMode', 'debug'],
-  alias: {},
-  default: envDefaults
-})
+async function run (): Promise<void> {
+  let config: ServerConfigParams
+  let web3provider
+  try {
+    const conf = await parseServerConfig(process.argv.slice(2), process.env)
+    if (conf.ethereumNodeUrl == null) {
+      error('missing ethereumNodeUrl')
+    }
+    web3provider = new Web3.providers.HttpProvider(conf.ethereumNodeUrl)
+    config = await resolveServerConfig(conf, web3provider) as ServerConfigParams
+  } catch (e) {
+    error(e.message)
+  }
+  const { devMode, workdir } = config
+  if (devMode) {
+    if (fs.existsSync(`${workdir}/${TXSTORE_FILENAME}`)) {
+      fs.unlinkSync(`${workdir}/${TXSTORE_FILENAME}`)
+    }
+  }
 
-if (argv._.length > 0) error(`unknown extra params: ${argv._.toString()}`)
+  const managerKeyManager = new KeyManager(1, workdir + '/manager')
+  const workersKeyManager = new KeyManager(1, workdir + '/workers')
+  const txStoreManager = new TxStoreManager({ workdir })
+  const gasPriceFactor = (config.gasPricePercent + 100) / 100
+  const { relayHubAddress, baseRelayFee, pctRelayFee, port, url } = config
+  const contractInteractor = new ContractInteractor(web3provider, configureGSN({}))
```

Comment:
I guess that you merged since you dropped `new ContractInteractor(web3provider,
  const httpServer = new HttpServer(port, relay)
  configureGSN({ relayHubAddress }))`

Reply:
fixed (last week...)